### PR TITLE
Shading work

### PR DESCRIPTION
**Pull request by est77**
- Added Blinn OSL closure. Removed Blinn special cases in OSL headers and shaders.
- Removed compatibility with OSL 1.6.
- Added anisotropic STD. Fixed some numerical issues.
- Added STD to BSDFs and OSL closures.

### CODE REVIEW

File: src/appleseed/renderer/modeling/bsdf/blinnbrdf.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        }
+
+        void operator()(
+            const foundation::Vector3f& o,
+            const foundation::Vector3f& h,
+            const foundation::Vector3f& n,
+            Spectrum&                   value) const
+        {
+            float f;
+            foundation::fresnel_reflectance_dielectric(
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/blinnbrdf.cpp]: Remove redundant namespace qualifiers.

File: src/appleseed/renderer/modeling/bsdf/blinnbrdf.cpp [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+            // No reflection below the shading surface.
+            const Vector3f& n = shading_basis.get_normal();
+            const float cos_in = dot(incoming, n);
+            const float cos_on = dot(outgoing, n);
+            if (cos_in < 0.0f || cos_on < 0.0f)
+                return 0.0f;
+
+            const InputValues* values = static_cast<const InputValues*>(data);
+            FresnelFun f(values->m_precomputed.m_outside_ior / values->m_ior);
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/blinnbrdf.cpp]: Make const.

File: src/appleseed/renderer/modeling/bsdf/glassbsdf.cpp [hunk 3]
```diff
@@ -72,10 +72,6 @@ namespace
     //
     // Glass BSDF.
     //
-    //    A future version of this BSDF will support multiple scattering.
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/glassbsdf.cpp]: So that's no longer a goal?

File: src/appleseed.shaders/src/alshaders/alSurface.osl [hunk 4]
```diff
@@ -468,10 +468,11 @@ shader alSurface
             transmissionNormal,
             vector(0),
             transmissionColor,
-            color(0), // transmission only.
+            color(0),                       // Transmission only.
```
Comment (dictoon) [inline src/appleseed.shaders/src/alshaders/alSurface.osl]: Our convention for end-of-line comments is no capital, no period:

    // transmission only

File: src/appleseed/renderer/modeling/bsdf/glassbsdf.cpp (hunk 3 above)
Comment (est77) [inline src/appleseed/renderer/modeling/bsdf/glassbsdf.cpp]: STD does not work with the multiple scattering model, at least for now.

**Pull request merged**